### ISSUE TITLE
Feature Request: HTTP Resolver to Private ELB or VPC Link support

### ISSUE THREAD

**Issue by darrenfurr** [aws/aws-appsync-community#81]
We're looking to invoke an AppSync HTTP resolver to an internal NLB. Currently, we do this with a VPC Link in API Gateway. We could tie the HTTP Resolver to the API Gateway, but that seems like overkill.

Are there plans to support internal endpoints?
- Lambdas inside a VPC
- Internal ELB(s)

I tried following the invoking AWS Services in the HTTP Resolver tutorial, but had no luck.
https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-http-resolvers.html

**Comment by parkerfath** [comment]
You are correct that these features are not currently supported. This is helpful feedback, though, in defining our future roadmap. Thanks for the feedback!

In the meantime, you can use a Lambda to connect to data sources inside a VPC, or continue to use the API Gateway workaround you mentioned.

**Comment by darrenfurr** [comment]
@tinnou - Can this be prioritized as a feature request? I'm looking to put lambdas inside a VPC & have AppSync connect to them.

**Comment by ezalorsara** [comment]
Good Day everyone, 
Is there any update about this? any solutions, we are running elasticsearch inside  vpc and we want to connect appsync to elasticsearch.  Thanks

**Comment by AaronHarris** [comment]
Best option at this point is to proxy the ELB request through a Lambda function

**Comment by somq** [comment]
Any update?

**Comment by nicolobissaccowarda** [comment]
we need this feature. 
Any news?

**Comment by bcs-gbl** [comment]
+1

**Comment by mkp-ebsco** [comment]
Any updates on this feature

**Comment by naderfares** [comment]
This is a really basic feature! I don't know how they didn't provide it yet. All my microservices are running in private subnets as they should be.

**Comment by jtich** [comment]
+1

**Comment by dvsingh9** [comment]
waiting....

**Comment by zebrometer** [comment]
+1

**Comment by bharathEF** [comment]
+1

**Comment by overgoro56** [comment]
+1

**Comment by Paco-lo** [comment]
Needed for any proffesional solution. 
This could limit the spread of AppSync solutions.
Thanks!

**Comment by tri-nguyen-nfq** [comment]
+1 Do we have any update/ plan for this feature?

**Comment by prasaadk** [comment]
+1

**Comment by kauefelipe-finck** [comment]
+1

**Comment by ManuPrivacyNinja** [comment]
Good luck getting those guys to do anything. Pretty sure we're stuck with using a Lambda, or just putting all the resources outside of the VPC, because, quite evidently it's so much more secure...

**Comment by mmierzwa** [comment]
+1

**Comment by Squinelato** [comment]
+1

**Comment by Viet1612** [comment]
+1

**Comment by tonistayhungrystayfoolish** [comment]
If that comes true I will adopt Appsync as a solution right now is not serious for a prod environment without this feature I have all my backend exposed in a API Gateway inside the VPC and doing a lambda resolver is not viable because of the increase in the bill + latency, please reconsider giving access to a VPC to a HTTP Datasource. Thank you.

**Comment by mmierzwa** [comment]
Are there any updates in this topic?

**Comment by mmierzwa** [comment]
bump 🙂

**Comment by yoni536** [comment]
any updates?

**Comment by tvmendoza** [comment]
The AppSync team is currently working on this feature. We will update as we get closer to launch.

**Comment by bcs-gbl** [comment]
> The AppSync team is currently working on this feature. We will update as we get closer to launch.

Great news! Thank you!

**Comment by jbschooley** [comment]
Will OpenSearch clusters inside a VPC be supported as well or just HTTP?

**Comment by glc-omid** [comment]
> The AppSync team is currently working on this feature. We will update as we get closer to launch. 

@tvmendoza 

Any progress after 5 months? 

Thanks in advance

**Comment by onlybakam** [comment]
Thanks for reaching. There is currently no update. We will keep this ticket open and updated as we have more info to share.

**Comment by glc-omid** [comment]
> Thanks for reaching. There is currently no update. We will keep this ticket open and updated as we have more info to share.

@onlybakam 
Thanks for your response, we move to other services , thanks ,

**Comment by billfine** [comment]
I can confirm that this feature is under active development and a priority for the AppSync team.

**Comment by themondays** [comment]
Just let me know if you need more hands. Sounds like AWS investing millions only for this feature 💪
However, +1 :)

**Comment by TamilCodes4AWS** [comment]
+1

**Comment by thecaddy** [comment]
+1

**Comment by poweredbyjpr** [comment]
+1

**Comment by ocasta** [comment]
+1

**Comment by uclaeamsavino** [comment]
Also referenced in issue #117

**Comment by JordanMRichards** [comment]
+1

**Comment by abisov** [comment]
+1

**Comment by nachovalera** [comment]
+1

**Comment by jack-arden-13** [comment]
+1

**Comment by beatrizdemiguelperez** [comment]
+1

**Comment by felipemengatto** [comment]
+1

**Comment by beatrizdemiguelperez** [comment]
+1

**Comment by narfasec** [comment]
+1

**Comment by themondays** [comment]
-1 (after 6 years seeing no progress)
